source/val/validation_state.h

```diff
@@ -1023,6 +1028,11 @@ class ValidationState_t {
   // TypePass.
   std::unordered_set<uint32_t> pointer_to_storage_image_;
 
+  /// Maps an id of DebugSource to a vector that contains the length of each
+  /// line side of it. (Also will have the DebugSourceContinued source included)
+  std::unordered_map<uint32_t, std::vector<uint32_t>> debug_source_line_length_;
+  uint32_t last_debug_source_id_;
```

Comment:
Prefer dropping this.

Reply:
So if there is

```
  %source_text_0 = OpString "
        lots of source
        ....
        hit string max"
  %source_text_1 = OpString "
        rest of the source"
        
%debug_source   = OpExtInst %void %import DebugSource %file_name %source_text_0
%debug_source_c = OpExtInst %void %import DebugSourceContinued %source_text_1
```

I need a way to know to append the `%source_text_1` to `source_text_0`

There is no way to connect (to my knowledge) the `DebugSourceContinued` back to `DebugSource` other then it is the previous instruction, and since we loop an instruction at the time, we need some value somewhere to mark the last `DebugSource`

Reply:
I thought I had written a comment on this, but I'm not sure what happened to it. Basically in the builder you'd do the following:
```
uint32_t source_id = inst->id();
while (ext_inst_index != NonSemanticShaderDebugInfo100DebugSource) {
  auto prev_index = inst - &_.ordered_instructions()[0];
  auto prev_inst = _.ordered_instructions()[prev_index];
  source_id = prev_inst->id();
  ext_inst_index = prev_inst->GetOperandAs<uint32_t>(3);
}

// Then use source_id as the entry into the map.
```

Reply:
ok nice, didn't realize there was a `prev_inst` flow doing `inst - &_.ordered_instructions()[0];` but ya... this is way better then holding a boolean around, will update

Reply:
update - way nicer, thanks